api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.dhsc.htbhf.hmrc.model.EligibilityRequest;
+import uk.gov.dhsc.htbhf.hmrc.model.EligibilityResponse;
+import uk.gov.dhsc.htbhf.hmrc.model.EligibilityStatus;
+import uk.gov.dhsc.htbhf.hmrc.model.PersonDTO;
+import uk.gov.dhsc.htbhf.hmrc.repository.HouseholdRepository;
+
+import java.net.URI;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.dhsc.htbhf.hmrc.model.EligibilityStatus.ELIGIBLE;
+import static uk.gov.dhsc.htbhf.hmrc.model.EligibilityStatus.NOMATCH;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.EligibilityRequestTestFactory.aValidEligibilityRequest;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.EligibilityRequestTestFactory.anEligibilityRequestWithPerson;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.EligibilityResponseTestFactory.aValidEligibilityResponseBuilder;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.EligibilityResponseTestFactory.anEligibilityResponse;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.HMRCEligibilityRequestTestDataFactory.aValidHMRCEligibilityRequest;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.HouseholdTestDataFactory.aHousehold;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.HouseholdTestDataFactory.aHouseholdWithChildrenAged6and24months;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.PersonDTOTestFactory.aValidPersonBuilder;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.TestConstants.HOMER_NINO;
+import static uk.gov.dhsc.htbhf.hmrc.testhelper.TestConstants.HOUSEHOLD_INDENTIFIER;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class HMRCIntegrationTests {
+
+    private static final URI ENDPOINT = URI.create("/v1/hmrc/eligibility");
+
+    private static final String HMRC_URL = "http://localhost:8130/v1/hmrc/benefits";
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private HouseholdRepository householdRepository;
+
+    @MockBean
+    private RestTemplate restTemplateWithIdHeaders;
+
+    @Test
+    void shouldReturnEligibilityResponseWhenNotInDatabase() {
+        //Given
+        ResponseEntity<EligibilityResponse> hmrcEligibilityResponse = new ResponseEntity<>(anEligibilityResponse(), OK);
+        given(restTemplateWithIdHeaders.postForEntity(anyString(), any(), eq(EligibilityResponse.class))).willReturn(hmrcEligibilityResponse);
+
+        //When
+        ResponseEntity<EligibilityResponse> response = restTemplate.exchange(buildRequestEntity(aValidEligibilityRequest()), EligibilityResponse.class);
```

Comment:
You should use the `callService` method here too.

Reply:
Refactored to use `callService()`